### PR TITLE
EZP-25140: Add the possibility to override files when copying configuration files

### PR DESCRIPTION
**Pull request by clash82**
Link to Jira task: https://jira.ez.no/browse/EZP-25140

### CODE REVIEW

**Comment by clash82** [comment]
ping @bdunogier @andrerom

File: eZ/Bundle/PlatformInstallerBundle/src/Installer/DbBasedInstaller.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ public function setOutput($output)
     protected function copyConfigurationFile($source, $target)
     {
         $fs = new Filesystem();
-        $fs->copy($source, $target);
+        $fs->copy($source, $target, true);
```
Comment (bdunogier) [inline eZ/Bundle/PlatformInstallerBundle/src/Installer/DbBasedInstaller.php]: I'd maybe add the copy option to the method, as an option argument that defaults to false (it was the default until then).

What do you think ?

**Comment by andrerom** [comment]
see https://github.com/ezsystems/DemoBundle/pull/191

File: eZ/Bundle/PlatformInstallerBundle/src/Installer/DbBasedInstaller.php (hunk 1 above)
Comment (andrerom) [inline eZ/Bundle/PlatformInstallerBundle/src/Installer/DbBasedInstaller.php]: See https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/Filesystem/Filesystem.php#L38

Imho If we add the option we should go for true by default.

Comment (clash82) [inline eZ/Bundle/PlatformInstallerBundle/src/Installer/DbBasedInstaller.php]: I like the idea but I think we should stay with `false` as a default to be more compatible with `copy` option. But I might be wrong? :)

Comment (clash82) [inline eZ/Bundle/PlatformInstallerBundle/src/Installer/DbBasedInstaller.php]: Updated.

**Comment by bdunogier** [comment]
I did meant false, actually. Good.

**Comment by bdunogier** [comment]
On second thought... it is such a good idea to change the method's signature ? Given how simple the installer is, it would make sense to just have it overwrite.

What about no argument, always overwrite, and specify it in the method's phpdoc ?

**Comment by clash82** [comment]
> What about no argument, always overwrite, and specify it in the method's phpdoc ?

This was in the first version :) Updated, once again...

**Comment by bdunogier** [comment]
> This was in the first version :) Updated, once again...

I know... sorry about that. It will happen again, it takes time to put your head in the right context :-)

**Comment by lserwatka** [comment]
@bdunogier do you approve this change in the current state?

**Comment by andrerom** [comment]
To answer your question..

**Comment by clash82** [comment]
Thank you :)

**Pull request merged**